### PR TITLE
SPECIES NUKING 2023: Nukes species_traits, good night sweet prince

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

IT'S OVER.

## Why It's Good For The Game

Species traits are a relic of a time before the trait system was added to generalize this kind of behavior.
They are clunky and overall less useful than inherent_traits - Converting these makes it easier to make these behaviors modular and usable not only by species.

## Changelog

:cl:
refactor: A significant species refactor happened, report any issues on the github.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/init_signals.dm [hunk 1]
```diff
@@ -2,12 +2,23 @@
 /mob/living/carbon/register_init_signals()
 	. = ..()
 
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_AGENDER), PROC_REF(on_nobreath_trait_gain))
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/init_signals.dm:5]:
```suggestion
	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_AGENDER), PROC_REF(on_agender_trait_gain))
```
Also, doesn't this need a trait loos counterpart too?

File: code/modules/client/preferences/species_features/mutants.dm [hunk 2]
```diff
@@ -2,7 +2,15 @@
...
 	category = PREFERENCE_CATEGORY_SECONDARY_FEATURES
-	relevant_species_trait = MUTCOLORS
+	relevant_inherent_trait = TRAIT_MUTANT_COLORS
+
+/datum/preference/color/mutant_color/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
+	var/datum/species/species = new species_type
```
Comment (Ghommie) [inline code/modules/client/preferences/species_features/mutants.dm:12]:
```suggestion
	var/datum/species/species = GLOB.species_list[initial(species_type.id)]
```

File: code/modules/mob/living/carbon/init_signals.dm (hunk 1 above)
Comment (ChungusGamer666) [reply code/modules/mob/living/carbon/init_signals.dm:5]: I'll implement it soon, thanks Ghom

**Comment by ChungusGamer666** [comment]
@Ghommie doing what you told me to do in the review creates runtimes, we cant do it for whatever god damn reason

**Comment by ChungusGamer666** [comment]
you have to be kidding me

File: code/modules/antagonists/nightmare/nightmare_species.dm [hunk 3]
```diff
@@ -7,14 +7,8 @@
...
 	changesource_flags = MIRROR_BADMIN | WABBAJACK | MIRROR_PRIDE
 	no_equip_flags = ITEM_SLOT_MASK | ITEM_SLOT_OCLOTHING | ITEM_SLOT_GLOVES | ITEM_SLOT_FEET | ITEM_SLOT_ICLOTHING | ITEM_SLOT_SUITSTORE
-	species_traits = list(
-		NO_UNDERWEAR,
-		NO_DNA_COPY,
-		NOTRANSSTING,
-	)
 	inherent_traits = list(
-		TRAIT_ADVANCEDTOOLUSER,
-		TRAIT_CAN_STRIP,
```
Comment (Time-Green) [inline code/modules/antagonists/nightmare/nightmare_species.dm:17]: These seem to have been removed?

File: code/modules/client/preferences/species_features/mutants.dm [hunk 4]
```diff
@@ -2,7 +2,15 @@
...
+	return !(TRAIT_FIXED_MUTANT_COLORS in species.inherent_traits)
```
Comment (Time-Green) [inline code/modules/client/preferences/species_features/mutants.dm:13]: Couldnt you do initial(species.inherent_traits) ? Then you can avoid instancing the species type

File: code/modules/client/preferences/underwear_color.dm [hunk 5]
```diff
@@ -3,13 +3,13 @@
...
-	target.underwear_color = value
-
 /datum/preference/color/underwear_color/is_accessible(datum/preferences/preferences)
 	if (!..(preferences))
 		return FALSE
 
 	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
 	var/datum/species/species = new species_type
-	return !(NO_UNDERWEAR in species.species_traits)
+	return !(TRAIT_NO_UNDERWEAR in species.inherent_traits)
```
Comment (Time-Green) [inline code/modules/client/preferences/underwear_color.dm:12]: initial?

**Review by Time-Green — CHANGES_REQUESTED** [review]
Great work! Just checking some stuff

File: code/modules/antagonists/nightmare/nightmare_species.dm (hunk 3 above)
Comment (ChungusGamer666) [reply code/modules/antagonists/nightmare/nightmare_species.dm:17]: they are in the brain organ itself already, should not be here!

File: code/modules/client/preferences/species_features/mutants.dm (hunk 4 above)
Comment (ChungusGamer666) [reply code/modules/client/preferences/species_features/mutants.dm:13]: You can't use initial on lists. I tried using the GLOB.species_list datums but it broke shit.

File: code/modules/client/preferences/underwear_color.dm (hunk 5 above)
Comment (ChungusGamer666) [reply code/modules/client/preferences/underwear_color.dm:12]: Not possible.

**Comment by ChungusGamer666** [comment]
Also I'll convert skin tones to use a trait too, that boolean kind of bothers me.

File: code/modules/client/preferences/species_features/mutants.dm (hunk 4 above)
Comment (Time-Green) [reply code/modules/client/preferences/species_features/mutants.dm:13]: Ah, I thought it was bitflags

**Comment by Time-Green** [comment]
> Also I'll convert skin tones to use a trait too, that boolean kind of bothers me.

That’s cool, I’m just merging this first though before it grows out of control

File: code/modules/client/preferences/species_features/vampire.dm [hunk 6]
```diff
@@ -22,9 +22,6 @@
 GLOBAL_LIST_EMPTY(vampire_houses)
 
 /datum/preference/choiced/vampire_status/apply_to_human(mob/living/carbon/human/target, value)
-	if (!(relevant_species_trait in target.dna?.species.species_traits))
-		return
-
```
Comment (MrStonedOne) [inline code/modules/client/preferences/species_features/vampire.dm:27]: ??????

Comment (Ghommie) [reply code/modules/client/preferences/species_features/vampire.dm:27]: https://github.com/tgstation/tgstation/pull/76414

Comment (ChungusGamer666) [reply code/modules/client/preferences/species_features/vampire.dm:27]: my bad g

**Pull request merged**